projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -599,7 +607,37 @@ public function get_item_permissions_check( $request ) { //phpcs:ignore Variable
 	}
 
 	/**
-	 * Get status for all supported integrations.
+	 * Core logic for a single integration
+	 *
+	 * @param string $slug Integration slug.
+	 * @return array Integration status data.
+	 */
+	private function get_integration( $slug ) {
+		$config       = $this->get_supported_integrations()[ $slug ];
+		$status       = $config['type'] === 'plugin'
+			? $this->get_plugin_status( $slug )
+			: $this->get_service_status( $slug );
+		$status['id'] = $slug;
+		return $status;
+	}
+
+	/**
+	 * REST callback for /integrations/{slug}
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function get_single_integration_status( $request ) {
+		$slug         = $request->get_param( 'slug' );
+		$integrations = $this->get_supported_integrations();
+		if ( ! isset( $integrations[ $slug ] ) ) {
+			return new \WP_Error( 'rest_integration_not_found', __( 'Integration not found.', 'jetpack-forms' ), array( 'status' => 404 ) );
+		}
+		return rest_ensure_response( $this->get_integration( $slug ) );
+	}
```

Comment:
This method is has the same name, but the logic is updated to relay on get_integration() and it is moved from below.